Tests fail with pygments-2.7.0+

Pygments 2.5.2 (as used in tox.ini) are pretty old. After upgrading to 2.7.0 or 2.7.1, I get the following test failures.

I can make a pull request updating the tests if you wish. Please just tell me if you prefer updating them purely for new pygments version, or if I should try to fit some regexps in to support both.

```
======================================================================
FAIL: test_codehilite_linenos_hl_lines (test_syntax.extensions.test_code_hilite.TestCodeHiliteClass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 289, in test_codehilite_linenos_hl_lines
    self.assertOutputEquals('line 1\nline 2\nline 3', expected, lang='text', linenos='inline', hl_lines=[1, 3])
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 43, in assertOutputEquals
    self.assertMultiLineEqual(output.strip(), expected)
AssertionError: '<div[58 chars]inenos">1</span><span class="hll">line 1\n</sp[119 chars]div>' != '<div[58 chars]ineno">1 </span><span class="hll">line 1\n</sp[119 chars]div>'
- <div class="codehilite"><pre><span></span><code><span class="linenos">1</span><span class="hll">line 1
?                                                                    -
+ <div class="codehilite"><pre><span></span><code><span class="lineno">1 </span><span class="hll">line 1
?                                                                       +
- </span><span class="linenos">2</span>line 2
?                           -
+ </span><span class="lineno">2 </span>line 2
?                              +
- <span class="linenos">3</span><span class="hll">line 3
?                    -
+ <span class="lineno">3 </span><span class="hll">line 3
?                       +
  </span></code></pre></div>

======================================================================
FAIL: test_codehilite_linenos_inline (test_syntax.extensions.test_code_hilite.TestCodeHiliteClass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 216, in test_codehilite_linenos_inline
    self.assertOutputEquals('plain text', expected, lang='text', linenos='inline')
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 43, in assertOutputEquals
    self.assertMultiLineEqual(output.strip(), expected)
AssertionError: '<div[43 chars]><span class="linenos">1</span>plain text\n</code></pre></div>' != '<div[43 chars]><span class="lineno">1 </span>plain text\n</code></pre></div>'
- <div class="codehilite"><pre><span></span><code><span class="linenos">1</span>plain text
?                                                                    -
+ <div class="codehilite"><pre><span></span><code><span class="lineno">1 </span>plain text
?                                                                       +
  </code></pre></div>

======================================================================
FAIL: test_codehilite_linenos_linenospecial (test_syntax.extensions.test_code_hilite.TestCodeHiliteClass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 323, in test_codehilite_linenos_linenospecial
    self.assertOutputEquals('line 1\nline 2\nline 3', expected, lang='text', linenos='inline', linenospecial=2)
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 43, in assertOutputEquals
    self.assertMultiLineEqual(output.strip(), expected)
AssertionError: '<div[58 chars]inenos">1</span>line 1\n<span class="linenos s[77 chars]div>' != '<div[58 chars]ineno">1 </span>line 1\n<span class="lineno sp[77 chars]div>'
- <div class="codehilite"><pre><span></span><code><span class="linenos">1</span>line 1
?                                                                    -
+ <div class="codehilite"><pre><span></span><code><span class="lineno">1 </span>line 1
?                                                                       +
- <span class="linenos special">2</span>line 2
?                    -
+ <span class="lineno special">2 </span>line 2
?                               +
- <span class="linenos">3</span>line 3
?                    -
+ <span class="lineno">3 </span>line 3
?                       +
  </code></pre></div>

======================================================================
FAIL: test_codehilite_linenos_linenostep (test_syntax.extensions.test_code_hilite.TestCodeHiliteClass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 306, in test_codehilite_linenos_linenostep
    self.assertOutputEquals('line 1\nline 2\nline 3', expected, lang='text', linenos='inline', linenostep=2)
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 43, in assertOutputEquals
    self.assertMultiLineEqual(output.strip(), expected)
AssertionError: '<div[58 chars]inenos"> </span>line 1\n<span class="linenos">[69 chars]div>' != '<div[58 chars]ineno">  </span>line 1\n<span class="lineno">2[69 chars]div>'
- <div class="codehilite"><pre><span></span><code><span class="linenos"> </span>line 1
?                                                                    -
+ <div class="codehilite"><pre><span></span><code><span class="lineno">  </span>line 1
?                                                                      +
- <span class="linenos">2</span>line 2
?                    -
+ <span class="lineno">2 </span>line 2
?                       +
- <span class="linenos"> </span>line 3
?                    -
+ <span class="lineno">  </span>line 3
?                      +
  </code></pre></div>

======================================================================
FAIL: test_codehilite_linenostart (test_syntax.extensions.test_code_hilite.TestCodeHiliteClass)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 271, in test_codehilite_linenostart
    self.assertOutputEquals('plain text', expected, lang='text', linenos='inline', linenostart=42)
  File "/tmp/markdown/tests/test_syntax/extensions/test_code_hilite.py", line 43, in assertOutputEquals
    self.assertMultiLineEqual(output.strip(), expected)
AssertionError: '<div[44 chars]<span class="linenos">42</span>plain text\n</code></pre></div>' != '<div[44 chars]<span class="lineno">42 </span>plain text\n</code></pre></div>'
- <div class="codehilite"><pre><span></span><code><span class="linenos">42</span>plain text
?                                                                    -
+ <div class="codehilite"><pre><span></span><code><span class="lineno">42 </span>plain text
?                                                                        +
  </code></pre></div>

======================================================================
FAIL: testFencedLanguageAttrNoclasses (test_syntax.extensions.test_fenced_code.TestFencedCode)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/tmp/markdown/tests/test_syntax/extensions/test_fenced_code.py", line 641, in testFencedLanguageAttrNoclasses
    self.assertMarkdownRenders(
  File "/tmp/markdown/markdown/test_tools.py", line 65, in assertMarkdownRenders
    self.assertMultiLineEqual(output, expected)
AssertionError: '<div[79 chars] 125%; margin: 0;"><span></span><code><span st[83 chars]div>' != '<div[79 chars] 125%"><span></span><code><span style="color: [71 chars]div>'
- <div class="python codehilite" style="background: #f8f8f8"><pre style="line-height: 125%; margin: 0;"><span></span><code><span style="color: #408080; font-style: italic"># Some python code</span>
?                                                                                         ------------
+ <div class="python codehilite" style="background: #f8f8f8"><pre style="line-height: 125%"><span></span><code><span style="color: #408080; font-style: italic"># Some python code</span>
  </code></pre></div>

----------------------------------------------------------------------
Ran 877 tests in 3.770s

FAILED (failures=6, skipped=21)
```

--- a/tests/test_syntax/extensions/test_code_hilite.py
+++ b/tests/test_syntax/extensions/test_code_hilite.py
@@ -205,7 +205,7 @@ class TestCodeHiliteClass(TestCase):
     def test_codehilite_linenos_inline(self):
         if has_pygments:
             expected = (
-                '<div class="codehilite"><pre><span></span><code><span class="lineno">1 </span>plain text\n'
+                '<div class="codehilite"><pre><span></span><code><span class="linenos">1</span>plain text\n'
                 '</code></pre></div>'
             )
         else:
@@ -259,7 +259,7 @@ class TestCodeHiliteClass(TestCase):
     def test_codehilite_linenostart(self):
         if has_pygments:
             expected = (
-                '<div class="codehilite"><pre><span></span><code><span class="lineno">42 </span>plain text\n'
+                '<div class="codehilite"><pre><span></span><code><span class="linenos">42</span>plain text\n'
                 '</code></pre></div>'
             )
         else:
@@ -274,9 +274,9 @@ class TestCodeHiliteClass(TestCase):
         if has_pygments:
             expected = (
                 '<div class="codehilite"><pre><span></span><code>'
-                '<span class="lineno">1 </span><span class="hll">line 1\n'
-                '</span><span class="lineno">2 </span>line 2\n'
-                '<span class="lineno">3 </span><span class="hll">line 3\n'
+                '<span class="linenos">1</span><span class="hll">line 1\n'
+                '</span><span class="linenos">2</span>line 2\n'
+                '<span class="linenos">3</span><span class="hll">line 3\n'
                 '</span></code></pre></div>'
             )
         else:
@@ -291,9 +291,9 @@ class TestCodeHiliteClass(TestCase):
     def test_codehilite_linenos_linenostep(self):
         if has_pygments:
             expected = (
-                '<div class="codehilite"><pre><span></span><code><span class="lineno">  </span>line 1\n'
-                '<span class="lineno">2 </span>line 2\n'
-                '<span class="lineno">  </span>line 3\n'
+                '<div class="codehilite"><pre><span></span><code><span class="linenos"> </span>line 1\n'
+                '<span class="linenos">2</span>line 2\n'
+                '<span class="linenos"> </span>line 3\n'
                 '</code></pre></div>'
             )
         else:
@@ -308,9 +308,9 @@ class TestCodeHiliteClass(TestCase):
     def test_codehilite_linenos_linenospecial(self):
         if has_pygments:
             expected = (
-                '<div class="codehilite"><pre><span></span><code><span class="lineno">1 </span>line 1\n'
-                '<span class="lineno special">2 </span>line 2\n'
-                '<span class="lineno">3 </span>line 3\n'
+                '<div class="codehilite"><pre><span></span><code><span class="linenos">1</span>line 1\n'
+                '<span class="linenos special">2</span>line 2\n'
+                '<span class="linenos">3</span>line 3\n'
                 '</code></pre></div>'
             )
         else:

--- a/tests/test_syntax/extensions/test_fenced_code.py
+++ b/tests/test_syntax/extensions/test_fenced_code.py
@@ -629,7 +629,7 @@ class TestFencedCode(TestCase):
         if self.has_pygments:
             expected = (
                 '<div class="python codehilite" style="background: #f8f8f8">'
-                '<pre style="line-height: 125%"><span></span><code>'
+                '<pre style="line-height: 125%; margin: 0;"><span></span><code>'
                 '<span style="color: #408080; font-style: italic"># Some python code</span>\n'
                 '</code></pre></div>'
             )

--- a/tox.ini
+++ b/tox.ini
@@ -14,7 +14,7 @@ commands =
 # Run tests with pygments installed (override deps only).
 deps =
     pytidylib
-    pygments==2.5.2
+    pygments==2.7.1
 
 [testenv:flake8]
 deps = flake8

